plugins/available/goenv.plugin.bash

```diff
@@ -0,0 +1,21 @@
+cite about-plugin
+about-plugin 'Init goenv, if installed. Plays nicely with package managers'
+
+# Try to set GOENV_ROOT if not already set
+if [ -z "${GOENV_ROOT}" ]; then
+	# Bail if cannot be set
+	if [ ! -d "$HOME/.goenv" ]; then
```

Comment:
As I already said in #1510, `goenv` creates the directory for the user and this shouldn't be checked up-front.

Reply:
Indeed you are right - I just tested this locally and goenv will create ~/.goenv if GOENV_ROOT is not set and ~/.goenv does not already exist.

This means that we should generally *not* set GOENV_ROOT at all, since the default value might change within the tool and we couldn't know that. (this is probably what you were trying to say the whole time, but your PR setting it to `~/.goenv` by default muddied that idea)

One case where we *might* consider setting GOENV_ROOT is when all of these conditions are met:
1) `GOENV_ROOT` is not already set
2) `goenv` is not already on the path
3) `~/.goenv/bin/goenv` exists

In that case, if we plan to add `~/goenv/bin` to the path, we could consider it reasonable to set GOENV_ROOT to `~/.goenv` - Even if the tool's default value were to change, the user taking the time to install the binary there can be taken as a preference to use that folder as the GOENV_ROOT

I'll re-work my logic here to accommodate these ideas.  Thank you.

Reply:
Please stop deviating from the patterns set in the other env plugins.

Reply:
I can not stress enough that you are over complicating this. We set the value, if not already set, so it's explicit. If the user follows up and changes it, it will be caught at the next shell invocation. Further more, none of this is done if the tool isn't found in the first place.

Please stop deviating from patterns set out in the other env plugins that have been working for years. I think all of this would make more sense if you used any of these tools, but it sounds like you don't and continue to over-engineer your solutions. Please look at the other env plugins. They are extremely simple and that is preferred.

Reply:
It's also worth noting I'm already using my fork with this fixed how I suggested in #1510, as is my entire team, across SuSE, Ubuntu, and MacOS.